### PR TITLE
improve preboostable logic, feedback and layout

### PR DESCRIPTION
**Pull request by dkent600**
Resolves: https://github.com/daostack/alchemy/issues/739
...hopefully.

Refer to the logic described in the code comments added in this client lib PR:  https://github.com/daostack/client/pull/416/files

* eliminates some unnecessary constructions of BN
* adds '=' (for ">=") in text referring to `downStakeNeededToQueue`
* improves layout of amount-needed feedback in StakeButtons.scss
* doesn't show amount-needed feedback unless it is possible to boost, in StakeButtons.tsx
* in PreTransactionModal, when downstaking, don't show amount-to-boost, rather show amount-to-stake-to-unboost

### CODE REVIEW

**Comment by dkent600** [comment]
@jellegerbrandy I am premising my findings here based on the logic given in the new comments in https://github.com/daostack/alchemy/issues/739.  Until you've approved those comments, I don't think we should approve this PR.  So I'm blocking this one until that point.

File: src/components/Proposal/Staking/BoostAmount.tsx [hunk 1]
```diff
@@ -26,32 +25,34 @@ export default class BoostAmount extends React.Component<IProps, null> {
       [css.wrapper]: true,
       [css.detailView]: detailView,
     });
+    const nothing = <span className={css.boostedAmount}><b>&nbsp;</b></span>;
 
     return (
       <div className={wrapperClass}>
         {
-          proposal.stage === IProposalStage.Queued && !expired && proposal.upstakeNeededToPreBoost.gte(new BN(0)) ?
+          !expired ? ((proposal.stage === IProposalStage.Queued && proposal.upstakeNeededToPreBoost.gten(0)) ?
```
Comment (tibetsprague) [inline src/components/Proposal/Staking/BoostAmount.tsx]: I dont think this is right, cant you still boost/unboost a pre-boosted proposal that is expired? or do we want to turn that off

File: src/components/Proposal/Staking/BoostAmount.tsx [hunk 2]
```diff
@@ -26,32 +25,34 @@ export default class BoostAmount extends React.Component<IProps, null> {
...
+                &gt; {formatTokens(proposal.upstakeNeededToPreBoost, "GEN")} on Pass to boost
               </b>
             </span>
-            : proposal.stage === IProposalStage.PreBoosted && proposal.downStakeNeededToQueue.lte(new BN(0)) ?
+            : proposal.stage === IProposalStage.PreBoosted && proposal.downStakeNeededToQueue.lten(0) ?
               <span className={css.boostedAmount}>
                 <b>
                   {detailView ? <img src="/assets/images/Icon/Boost-slate.svg" /> : ""}
-                &gt; {formatTokens(proposal.downStakeNeededToQueue.abs(), "GEN")} Pass to stay boosted
+                &gt; {formatTokens(proposal.downStakeNeededToQueue.abs(), "GEN")} on Pass to stay boosted
```
Comment (tibetsprague) [inline src/components/Proposal/Staking/BoostAmount.tsx]: does this fit?

Comment (dkent600) [reply src/components/Proposal/Staking/BoostAmount.tsx]: yes

File: src/components/Proposal/ActionButton.tsx [hunk 3]
```diff
@@ -230,13 +230,13 @@ class ActionButton extends React.Component<IProps, IState> {
           /> : ""
         }
 
-        { proposalState.stage === IProposalStage.Queued && proposalState.upstakeNeededToPreBoost.lt(new BN(0)) ?
+        { proposalState.stage === IProposalStage.Queued && proposalState.upstakeNeededToPreBoost.ltn(0) ?
```
[the remark this answers is not in the document]
Comment (dkent600) [reply src/components/Proposal/ActionButton.tsx]: has always worked for me.  testing the functionality seems to work

File: src/components/Proposal/Staking/BoostAmount.tsx (hunk 1 above)
Comment (dkent600) [reply src/components/Proposal/Staking/BoostAmount.tsx]: I was misinterpreting what "expired" meant (forgetting that it often means expired only within the context of the current phase, but not necessarily that the proposal is dead, and not necessarily that the expired state is even represented yet in the blockchain).

So good catch.  I have reverted the change.

**Pull request merged**